### PR TITLE
fix(payments): record checkout "engage" event for consent checkbox

### PR DESCRIPTION
**Pull request by biancadanforth**
Because:
* We are seeing fewer "engage" events relative to "submit", even though "engage" occurs upstream in the conversion funnel.
* We were not recording an "engage" event for clicking the consent checkbox.
* This is the only form field present in all scenarios and checkout flows on the payments server, so it likely makes up the majority of the missed events across all flows.

This commit:

* Records the "fxa_pay_setup - engage" event when clicking the consent checkbox.

Closes #FXA-7941

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Comment by biancadanforth** [comment]
Okay I think I fixed the related [CI failure](https://app.circleci.com/pipelines/github/mozilla/fxa/43477/workflows/028ff11c-2d9c-44cf-86a8-6ad1d4076142/jobs/455134/tests).

TL;DR: we had some duplicate "view" and "engage" events. This isn't terrible, since we dedupe by flow ID at the analysis step, but ideally we should avoid unnecessary network calls. To fix this, I replaced the `useCallback` hook with the custom `useCallbackOnce` hook which ensures the callback only gets executed once.

**Review by IvoJP — APPROVED** [review]
